11-Backbonejs/02-Backbone-Views.md

```diff
@@ -2,145 +2,378 @@
 
-Drawing the view is done in the `render` function which determines how the view displays in the DOM.  Below the
+Views are created with the familiar `new` keyword. It will look something like this:
 
 ```javascript
-var TodoView = Backbone.View.extend({
-  initialize: function(){
-    this.title = "Study Ada Lovelace";
-    this.render();
+var taskView = new TaskView();
+```
+
+Right now our view isn't doing anything, so let's give it some data to keep track of. We'll create each view around one of the tasks in our list.
+
+```javascript
+var TaskView = Backbone.View.extend({
+  initialize: function(options) {
+    this.task = options.task;
   },
-  render: function(){
-    $(this.el).append("<ul> <li>" + this.name "</li> </ul>");
+
+  render: function() {
+    // Enable chained calls
     return this;
   }
 });
 
-$(document).ready(function(){
-	var myTodoView = new TodoView({ el: $("#todo") });
+$(document).ready(function() {
+  var taskView = new TaskView({task: taskData[0]});
+});
+```
+
+#### Render
+On to `render()`. This will be called whenever the DOM needs to be updated, and its role is to re-generate all the HTML associated with this view. The way to do this is by using `this.$el`, a jQuery element associated with this view. Update `render()` so it looks like this:
+
+```javascript
+render: function() {
+  var html = '<li class="task">';
+  html += '<h2>' + this.task.title + '</h2>';
+  html += '<p>' + this.task.description + '</p>';
+  html += '</li>';
+  this.$el.html($(html));
+
+  // Enable chained calls
+  return this;
+}
+```
+
+Then in `$(document).ready()`, call `render()`.
+
+```javascript
+$(document).ready(function() {
+  var taskListElement = $('.task-list');
+  var taskView = new TaskView({task: taskData[0]});
+  taskListElement.append(taskView.render().$el);
 });
 ```
 
-This will display in the browser:
+Notice that we use jQuery to select the task list in the DOM, and call `append()` to make our generated HTML show up. Notice also that, because we return `this` from `render()`, we can combine the call to `render()` and the call to `append()` in one line. This isn't strictly necessary - we could just as easily have said
 
+```javascript
+taskView.render();
+taskListElement.append(taskView.$el);
 ```
-*  Study Ada Lovelace
+
+Chaining on `render()` is a very common thing in Backbone, and much of the code you'll find on the internet assumes you're using it. It will be much less painful to get into the habit of always returning `this` from `render()` now.
+
+#### Many Views
+One last step remains. Right now we're only viewing one task, but we want to see all of them. This can be solved with a quick loop:
+
+```javascript
+$(document).ready(function() {
+  var taskListElement = $('.task-list');
+  var taskViews = []
+  taskData.forEach(function(task) {
```

Comment:
should we use a different variable in the iteration variable to distinguish between this one and the one associated with the `task:` param in the initializer?